src/renderer/services/eth2/client/eth2ApiClient/index.ts

```diff
@@ -0,0 +1,42 @@
+import {ICGEth2ValidatorApi, IValidatorBeaconClient, ICGEth2BeaconApi, ICGEth2NodeApi} from "../interface";
+import {IBeaconConfig} from "@chainsafe/lodestar-config";
+import {IEventsApi} from "@chainsafe/lodestar-validator/lib/api/interface/events";
+import {IEth2ChainHead} from "../../../../models/types/head";
+import {CgEth2BeaconApi} from "./cgEth2BeaconApi";
+import {HttpClient} from "../../../api";
+import {AbstractApiClient} from "@chainsafe/lodestar-validator/lib/api/abstract";
+import {WinstonLogger} from "@chainsafe/lodestar-utils";
+import {CgEth2ValidatorApi} from "./cgEth2ValidatorApi";
+import {CgEth2NodeApi} from "./cgEth2NodeApi";
+import {CgEth2EventsApi} from "./cgEth2EventsApi";
+
+export class CgEth2ApiClient extends AbstractApiClient implements IValidatorBeaconClient {
+    public validator: ICGEth2ValidatorApi;
+    public beacon: ICGEth2BeaconApi; //
+    public node: ICGEth2NodeApi;
+    public events: IEventsApi;
+
+    public url: string;
+
+    private readonly httpClient: HttpClient;
+    public constructor(config: IBeaconConfig, url: string) {
+        // TODO: logger: create new or get it from outside?
+        super(config, new WinstonLogger());
+        this.url = url;
+        this.httpClient = new HttpClient(url);
+
+        this.validator = new CgEth2ValidatorApi(config, this.httpClient);
+        this.beacon = new CgEth2BeaconApi(config, this.httpClient);
+        this.events = new CgEth2EventsApi(config, url);
+        this.node = new CgEth2NodeApi(config, this.httpClient);
+    }
+
+    public getVersion = async (): Promise<string> => {
+        throw new Error("Method 'getVersion' not implemented.");
+    };
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    public onNewChainHead = (callback: (head: IEth2ChainHead) => void): NodeJS.Timeout => {
+        throw new Error("Method 'onNewChainHead' not implemented.");
```

Comment:
implement using events api

Reply:
@BeroBurny I don't see this implemented anywhere?

Reply:
It is not used by any part of the code, if will be required it will be implemented correctly

Reply:
It's not used by validator but it's used by chainguardian. It's used to display beacon node current slot when syncing. You always used synced node so that part of code wasn't executed